### ISSUE TITLE
Not working with NodePort service

### ISSUE THREAD

**Issue by lesovsky** [kubernetes/kube-state-metrics#1139]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
The kube-state-metrics `/metrics` endpoint doesn't repond when using service with `type: NodePort`. Both http-metrics and telemetry services are not working. The `/` and `/healthz` endpoints respond correctly.

**What you expected to happen**:
The kube-state-metrics should respond correctly when doing requests to `/metrics` endpoint when using NodePort service (as it works with `/` and `/healthz`):

**How to reproduce it (as minimally and precisely as possible)**:
1. Clone the repo
2. Replace `clusterIP: None` with `type: NodePort` in `examples/standard/service.yaml` manifest and do `kubectl apply -f examples/standard`.
3. Check inside from kubernetes that kube-state-metrics works correctly and respond normally on `/metrics` requests (run a pod in the same namespace and make curl requests to kube-state-metrics' http-metrics and telemetry services). Everything should work normally here.
3. Check `nodePort` value in the output of `kubectl get svc -n kube-system` and make requests on those ports:
   - curl host:nodePort/ -- requesting `/` returns html (works as intended)
   - curl host:nodePort/healthz -- requesting `/healthz` returns `OK` (works as intended)
   - curl host:nodePort/metrics -- requesting `/metrics` hangs and not respond (not working, curl stats shows zeroes, no data received or transfered)
4. The same problem with telemetry port.

**Anything else we need to know?**:
The problem is the same when using nginx-ingress (with default service with `clusterIP: None`), but nginx resets the connection with `curl: (56) Recv failure: Connection reset by peer` after 60 seconds. When not wait the connection reset and press Ctrl+C, in the ingress log you can see the request is processed with 200 OK
```
192.168.196.0 - - [11/May/2020:17:02:16 +0000] "GET /metrics HTTP/1.1" 200 46838 "-" "curl/7.58.0" 89 68.757 [staging-kube-state-metrics-8080] [] 192.168.230.19:8080 308343 60.007 200 256359a7d4e7178d2148174e80b02876
```
I tried to increase various buffers in the ingress and instead of resets got timeout:
```
2020/05/11 17:02:07 [error] 3266#3266: *23486956 upstream timed out (110: Operation timed out) while reading upstream, client: 192.168.196.0, server: kube-state-metrics, request: "GET /metrics HTTP/1.1", upstream: "http://192.168.230.19:8080/metrics", host: "kube-state-metrics"
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T20:55:23Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```
- Kube-state-metrics image version:
Tried on 1.9.5, 1.9.4, 1.9.0, 1.8.0.

**Comment by brancz** [comment]
Can you connect to other things via node port? This doesn’t really sound like a kube state metrics problem.

**Comment by lesovsky** [comment]
Other things exposed using nodePort work correctly, but I've found the cause of my problem. I tried connecting to the host where no nodePort was opened. Everything is ok when connect is performed to proper host with exposed nodePort.
It seems, yesterday i was too tired and strongly confused by proper responding from '/' and '/healthz' endpoints. Today I've rechecked all the things and found where I was mistaken.
Pardon me for bothering
